internal/generator/generator.go

```diff
@@ -11,10 +11,17 @@ import (
 	"io"
 )
 
+type GenerateConfig struct {
+	Filter           map[string]struct{}
+	Override         map[string][]Mask
+	DirectiveMapName string
+	MatchFuncName    string
+}
```

Comment:
Nit: it would be good practice to add doc comments to this type to communicate what each of these config settings do.

Reply:
I see your point. But I thought the variable names here are clear? But it is just from my angle. If you felt unclear about any of them I can add comments. I'm trying to save some comments like this you've sent me before lol https://preview.redd.it/vs35fh1ruef51.jpg?width=640&crop=smart&auto=webp&s=3e1dd46e210f0499b9ac87187eefbde912bebdeb

Reply:
I think the names are decent, but here's an example that I think adds value.

```go
// DirectiveMapName is the name assigned to the variable containing the directives
// discovered in the NGINX module source. The variable name generally starts with a
// lowercase to avoid export. Users will use the generated function named by 
// MatchFuncName to validate the module directives in nginx configurations.
DirectiveMapName string
```

`MatchFuncName` would similarly be documented to communicate that it should be pascal case to be exported.